lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -417,6 +434,15 @@ void UsdUndoAssignNewMaterialCommand::execute()
                 markAsFailed();
                 return;
             }
+            // There might be some unassignable items in the selection list. Skip and warn.
+            // We know there is at least one assignable item found in the ContextOps resolver.
+            if (!BindMaterialUndoableCommand::CompatiblePrim(parentItem)) {
+                const std::string error = TfStringPrintf(
+                    "Assign new material: Skipping incompatible prim [%s] found in selection.",
+                    Ufe::PathString::string(parentItem->path()).c_str());
```

Comment:
Found in the unit tests where the first new material ended up in the selection list for the second assign new material call.